### PR TITLE
cleanup npm scripts

### PR DESCRIPTION
**Pull request by mcortesi**
### Description

Summary of changeS:

- Use build, lint and install as standard scripts for every package
- rename lint-checks to lint
- remove old dockerfile
- rename dockerfiles to use `Dockerfile` name
- Simplify circleci build
- add tslint to utils
- normalize script names
    - rename compile-typescript to build:ts
    - create build:gen for generated files
- don't compile typescript on lint. (build will do that)
- changes to celotool dockerfile
    - separate install & build to better cache hit ratio
    - fix dependencies (package.json, missing subdependency)
    - cleanup apt-get
    - reduce the number of docker layers
 - improve circleci workflow:
    - join build&install
    - e2e only run when protocol and contractkit succeeded
    - mobile build only runs after mobile test succeeded
    - tests only run if build was successfull
    - better cache strategy for node_modules
    - ignore optional deps in yarn (they take time and we don't use them)

### Tested

Locally run:
 - yarn install, yarn build, yarn test, yarn lint
 - CI passes

Something else? @ashishb

### CODE REVIEW

File: packages/contractkit/package.json [hunk 1]
```diff
@@ -7,13 +7,12 @@
...
-    "build": "yarn run clean && bin/build-sdk.js",
-    "compile-typescript": "tsc",
+    "build:env": "bin/build-sdk.js",
+    "build:ts": "tsc -p .",
     "clean": "rm -rf lib .artifacts contracts types lib",
     "lint": "tslint -c tslint.json --project tsconfig.json",
-    "lint-checks": "yarn run lint && yarn run build alfajores",
-    "postinstall": "yarn run build alfajores",
-    "prettify": "yarn run prettier --config ../../.prettierrc.js --write '{contracts,types}/**/*.+(ts|tsx|js|jsx)'",
+    "prebuild": "yarn clean",
```
Comment (jmrossy) [inline packages/contractkit/package.json]: Does prebuild get run automatically?

File: packages/mobile/package.json [hunk 2]
```diff
@@ -9,11 +9,10 @@
     "start:bg": "react-native start &",
     "start:android": "react-native run-android",
     "start:ios": "react-native run-ios",
-    "lint-checks": "yarn run lint && yarn run build:typescript",
     "lint": "tslint -c tslint.json --project tsconfig.json",
-    "build": "yarn run build:typescript && yarn run build:metro",
-    "build:sdk": "build-sdk",
-    "build:typescript": "tsc --noEmit",
+    "build": "yarn run build:ts && yarn run build:metro",
+    "build:sdk": "yarn --cwd ../contractkit build:env",
```
Comment (jmrossy) [inline packages/mobile/package.json]: is `env` in `build:env` a variable here?

File: packages/verifier/package.json [hunk 3]
```diff
@@ -9,9 +9,8 @@
     "dev:show-menu": "adb devices | grep '\t' | awk '{print $1}' | sed 's/\\s//g' | xargs -I {} adb -s {} shell input keyevent 82",
     "dev:reload": "adb shell input text \"RR\"",
     "dev:remote": "remotedev --hostname=localhost --port=8000",
-    "lint-checks": "yarn run lint && yarn run build:typescript",
     "lint": "tslint -c tslint.json --project tsconfig.json",
-    "build:typescript": "tsc",
+    "build": "tsc -p .",
```
Comment (jmrossy) [inline packages/verifier/package.json]: We should probably have --noEmit here like we have for the mobile project

**Review by jmrossy — COMMENTED** [review]
Amazing!! A few small comments below but this is a great cleanup!

**Review by yerdua — DISMISSED** [review]
Looks good to me! Though, since this changes the names of commands, it's worth sending out a message in slack/email in case this affects anyone's local workflows that rely on the old names.

File: packages/mobile/package.json (hunk 2 above)
Comment (mcortesi) [reply packages/mobile/package.json]: No, just the name of the npm script. You would still call `yarn build:sdk alfajores`.
I would prefer not to have a n npm script in mobile that call contractkit, but i saw that it was being use in a deploy script

Comment (jmrossy) [reply packages/mobile/package.json]: Ah okay. In that case, should it be build:for-env to be consistent with verification pool?

Comment (mcortesi) [reply packages/mobile/package.json]: yeah. i like it.

File: .circleci/config.yml [hunk 4]
```diff
@@ -7,6 +7,8 @@ defaults: &defaults
   working_directory: ~/app
   docker:
     - image: celohq/node8:gcloud-deps
+  environment:
```
Comment (ashishb) [inline .circleci/config.yml:10]: Add a comment that this was added to deal with `ENOMEM: not enough memory, `

**Review by ashishb — DISMISSED** [review]

**Review by timmoreton — APPROVED** [review]

**Review by timmoreton — APPROVED** [review]

**Pull request merged**